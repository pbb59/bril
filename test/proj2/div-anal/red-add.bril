# ARGS: diverge

main {
# init constants idxs
init:
  zero: int = const 0;
  one: int = const 1;
  two: int = const 2;
  three: int = const 3;

# simplest spatial redundancy elimination
pack:
  a0: int = const 7;
  a1: int = const 11;
  vec0: vector = s2vb a0;
  vec1: vector = s2vb a1;
add:
  vec2: vector = vadd vec0 vec1;
extract:
  a2: int = v2s vec2 zero;
  a3: int = v2s vec2 one;
  a4: int = v2s vec2 two;
  a5: int = v2s vec2 three;

# print results
print:
  print a2;
  print a3;
  print a4;
  print a5;
}
