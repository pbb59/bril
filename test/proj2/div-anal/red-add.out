init:
  in:  ∅
  out: ∅
pack:
  in:  ∅
  out: ∅
add:
  in:  ∅
  out: ∅
extract:
  in:  ∅
  out: ∅
print:
  in:  ∅
  out: ∅
